src/fights.ts

```diff
@@ -2174,10 +2180,83 @@ export function printEmbezzlerLog(): void {
     HIGHLIGHT
   );
 }
+type FreeKill = { source?: Item; skill: Skill; used: () => boolean };
+const freeKills: FreeKill[] = [
+  {
+    source: $item`The Jokester's gun`,
+    skill: $skill`Fire the Jokester's Gun`,
+    used: () => get("_firedJokestersGun"),
+  },
+  {
+    source: $item`Lil' Doctor™ bag`,
+    skill: $skill`Chest X-Ray`,
+    used: () => get("_chestXRayUsed") >= 3,
+  },
+  { skill: $skill`Shattering Punch`, used: () => get("_shatteringPunchUsed") >= 3 },
+  { skill: $skill`Gingerbread Mob Hit`, used: () => get("_gingerbreadMobHitUsed") },
+];
+const isAvailable = ({ source, skill, used }: FreeKill) => have(source ?? skill) && !used();
+const toRequirement = ({ source }: FreeKill) =>
+  source ? new Requirement([], { forceEquip: [source] }) : new Requirement([], {});
+function findFreeKill() {
+  return freeKills.find(isAvailable) ?? null;
+}
+
+function killRobortCreaturesForFree() {
+  if (!have($familiar`Robortender`)) return;
+  useFamiliar($familiar`Robortender`);
+
+  let freeKill = findFreeKill();
+  while (
+    freeKill &&
+    canAdv($location`The Copperhead Club`) &&
+    have($skill`Comprehensive Cartography`) &&
+    get("_monstersMapped") < 3
+  ) {
+    if (have($effect`Crappily Disguised as a Waiter`)) {
+      setChoice(855, 4);
+      adventureMacro($location`The Copperhead Club`, Macro.abort());
+    }
+    freeFightOutfit(toRequirement(freeKill));
+    withMacro(
+      Macro.skill(freeKill.skill),
+      () => {
+        mapMonster($location`The Copperhead Club`, $monster`Mob Penguin Capo`);
+        runCombat();
+      },
+      true
+    );
+    freeKill = findFreeKill();
```

Comment:
wait why are we doing this at the end again? is it just like... retabulating within the while to ensure the while is still active? that makes sense i guess. for some reason i would've thought the while logic would retabulate when it hits the end naturally but i am probably wrong.

Reply:
We initialize it before the first while, but update it after each subsequent while. We want to update it before the condition is called because we don't want to proceed if there is no available free fight now